File: Test/Desktop/OpenRiaServices.Common.DomainServices.Test/Shared/Mock.shared.cs
using System;
using System.Runtime.Serialization;

namespace TestDomainServices
{
    [Flags]
    [DataContract]
    public enum TestEnum
    {
        [EnumMember]
        Value0 = 0,

        [EnumMember]
        Value1 = 1,

        [EnumMember]
        Value2 = 2,

        [EnumMember]
        Value3 = 4
    }

    [AttributeUsage(AttributeTargets.All, AllowMultiple = false)]
    public class MockAttributeAllowOnce : Attribute
    {
        public MockAttributeAllowOnce(string value) { this.Value = value; }
        public string Value { get; private set; }
#if !SILVERLIGHT
        public override object TypeId { get { return Guid.NewGuid(); } }
#endif
    }

    [AttributeUsage(AttributeTargets.All, AllowMultiple = false)]
    public class MockAttributeAllowOnce_AppliedToInterfaceOnly : Attribute
    {
        public MockAttributeAllowOnce_AppliedToInterfaceOnly(string value) { this.Value = value; }
        public string Value { get; private set; }
#if !SILVERLIGHT
        public override object TypeId { get { return Guid.NewGuid(); } }
#endif
    }

    [AttributeUsage(AttributeTargets.Interface, AllowMultiple = false)]
    public class MockAttributeAllowOnce_InterfaceOnly : Attribute
    {
        public MockAttributeAllowOnce_InterfaceOnly(string value) { this.Value = value; }
        public string Value { get; private set; }
#if !SILVERLIGHT
        public override object TypeId { get { return Guid.NewGuid(); } }
#endif
    }

    [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
    public class MockAttributeAllowMultiple : Attribute
    {
        public MockAttributeAllowMultiple(string value) { this.Value = value; }
        public string Value { get; private set; }
#if !SILVERLIGHT
        public override object TypeId { get { return Guid.NewGuid(); } }
#endif
    }

    [AttributeUsage(AttributeTargets.Interface, AllowMultiple = true)]
    public class MockAttributeAllowMultiple_InterfaceOnly : Attribute
    {
        public MockAttributeAllowMultiple_InterfaceOnly(string value) { this.Value = value; }
        public string Value { get; private set; }
#if !SILVERLIGHT
        public override object TypeId { get { return Guid.NewGuid(); } }
#endif
    }
}

// Used to verify that the code-generator properly generates code for attributes defined in unrelated namespaces.
namespace CustomNamespace
{
    [AttributeUsage(AttributeTargets.All)]
    public class CustomAttribute : Attribute
    {
        //
    }
}

File: src/Test/Desktop/OpenRiaServices.Common.DomainServices.Test/Shared/Mock.shared.vb
Imports System
Imports System.ComponentModel.DataAnnotations
Imports System.Runtime.Serialization

Namespace TestDomainServices
    <Flags()> _
    <DataContract()> _
    Public Enum TestEnum
        <EnumMember()> _
        Value0 = 0

        <EnumMember()> _
        Value1 = 1

        <EnumMember()> _
        Value2 = 2

        <EnumMember()> _
        Value3 = 4
    End Enum

    <AttributeUsage(AttributeTargets.All, AllowMultiple:=False)> _
    Public Class MockAttributeAllowOnce
        Inherits Attribute
        Dim _value As String

        Public Sub New(ByVal value As String)
            Me.Value = value
        End Sub

        Public Property Value() As String
            Get
                Return Me._value
            End Get
            Set(ByVal value As String)
                Me._value = value
            End Set
        End Property
    End Class

    <AttributeUsage(AttributeTargets.All, AllowMultiple:=False)> _
    Public Class MockAttributeAllowOnce_AppliedToInterfaceOnly
        Inherits Attribute
        Dim _value As String

        Public Sub New(ByVal value As String)
            Me.Value = value
        End Sub

        Public Property Value() As String
            Get
                Return Me._value
            End Get
            Set(ByVal value As String)
                Me._value = value
            End Set
        End Property
    End Class

    <AttributeUsage(AttributeTargets.Interface, AllowMultiple:=False)> _
    Public Class MockAttributeAllowOnce_InterfaceOnly
        Inherits Attribute
        Dim _value As String

        Public Sub New(ByVal value As String)
            Me.Value = value
        End Sub

        Public Property Value() As String
            Get
                Return Me._value
            End Get
            Set(ByVal value As String)
                Me._value = value
            End Set
        End Property
    End Class

    <AttributeUsage(AttributeTargets.All, AllowMultiple:=True)> _
    Public Class MockAttributeAllowMultiple
        Inherits Attribute
        Dim _value As String

        Public Sub New(ByVal value As String)
            Me.Value = value
        End Sub

        Public Property Value() As String
            Get
                Return Me._value
            End Get
            Set(ByVal value As String)
                Me._value = value
            End Set
        End Property
    End Class

    <AttributeUsage(AttributeTargets.Interface, AllowMultiple:=True)> _
    Public Class MockAttributeAllowMultiple_InterfaceOnly
        Inherits Attribute
        Dim _value As String

        Public Sub New(ByVal value As String)
            Me.Value = value
        End Sub

        Public Property Value() As String
            Get
                Return Me._value
            End Get
            Set(ByVal value As String)
                Me._value = value
            End Set
        End Property
    End Class
End Namespace

' Used to verify that the code-generator properly generates code for attributes defined in unrelated namespaces.
Namespace CustomNamespace
    <AttributeUsage(AttributeTargets.All)> _
    Public Class CustomAttribute
        Inherits Attribute
    End Class
End Namespace